addons/website_sale_loyalty/models/sale_order.py

```diff
@@ -156,7 +156,17 @@ def get_promo_code_success_message(self, delete=True):
             request.session.pop('successful_code')
         return code
 
+    def _is_to_delete_discount(self, line, set_qty):
+        self.ensure_one()
+        return set_qty == 0 and line.coupon_id and line.reward_id and line.reward_id.reward_type == 'discount'
+
     def _cart_update(self, *args, **kwargs):
+        product_id, set_qty = kwargs['product_id'], kwargs.get('set_qty', None)
+
+        line = self.order_line.filtered(lambda l: l.product_id.id == product_id)
+        if self._is_to_delete_discount(line, set_qty):
```

Comment:
if it's possible to have several lines with the same product (it doesn't seem possible from my test, but maybe I have the wrong configuration), this could cause an issue

Reply:
It shouldn't be possible IMHO, but @Feyensv might know more... :thinking: